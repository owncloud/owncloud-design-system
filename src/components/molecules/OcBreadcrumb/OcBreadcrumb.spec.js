import { shallowMount } from "@vue/test-utils"
import Breadcrumb from "./OcBreadcrumb.vue"

const items = [
  { text: "First folder", to: { path: "folder" } },
  { text: "Subfolder", onClick: () => alert("Breadcrumb clicked!") },
  { text: "Deep", to: { path: "folder" } },
  { text: "Deeper ellipsize in responsive mode" },
]

describe("OcBreadcrumb", () => {
  it("sets correct variation", () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: {
        variation: "lead",
        items,
      },
    })

    expect(wrapper.props().variation).toMatch("lead")
    expect(wrapper.html()).toContain("oc-breadcrumb oc-breadcrumb-lead")
    expect(wrapper).toMatchSnapshot()
  })
  it("displays all items", () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: {
        items,
      },
    })

    expect(wrapper.findAll(".oc-breadcrumb-list-item").length).toBe(items.length)
    expect(wrapper).toMatchSnapshot()
  })
  it("displays context menu trigger when a slot is given", () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: {
        items,
      },
      slots: {
        contextMenu: "Example item",
      },
    })

    expect(wrapper.find("#oc-breadcrumb-contextmenu-trigger").exists()).toBe(true)
  })
  it("does not display context menu trigger when no slot given", () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: {
        items,
      },
    })

    expect(wrapper.find("#oc-breadcrumb-contextmenu-trigger").exists()).toBe(false)
  })
})
