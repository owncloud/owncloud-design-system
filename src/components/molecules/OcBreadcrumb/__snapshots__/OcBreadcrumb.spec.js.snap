// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcBreadcrumb displays all items 1`] = `
<nav class="oc-breadcrumb oc-breadcrumb-default">
  <ol class="oc-breadcrumb-list">
    <li class="oc-breadcrumb-list-item">
      <router-link-stub tag="a" to="[object Object]">
        First folder
      </router-link-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item">
      <oc-button-stub type="button" size="medium" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
        Subfolder
      </oc-button-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item">
      <router-link-stub tag="a" to="[object Object]">
        Deep
      </router-link-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item"><span aria-current="page" tabindex="-1">Deeper ellipsize in responsive mode</span>
      <!---->
    </li>
  </ol>
  <div class="oc-breadcrumb-drop"><label tabindex="0" class="oc-breadcrumb-drop-label"><span aria-current="page" class="oc-breadcrumb-drop-label-text">Deeper ellipsize in responsive mode</span>
      <oc-icon-stub name="expand_more" filltype="fill" accessiblelabel="Expand more" type="span" size="medium" variation="passive" class="oc-breadcrumb-drop-label-icon"></oc-icon-stub>
    </label>
    <oc-drop-stub dropid="oc-drop-2" toggle="" position="bottom-start" mode="click" paddingsize="medium" options="[object Object]">
      <ol>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            Deep
          </router-link-stub>
        </li>
        <li>
          <oc-button-stub type="button" size="medium" variation="passive" appearance="raw" justifycontent="left" gapsize="medium">
            Subfolder
          </oc-button-stub>
        </li>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            First folder
          </router-link-stub>
        </li>
      </ol>
    </oc-drop-stub>
  </div>
</nav>
`;

exports[`OcBreadcrumb sets correct variation 1`] = `
<nav class="oc-breadcrumb oc-breadcrumb-lead">
  <ol class="oc-breadcrumb-list">
    <li class="oc-breadcrumb-list-item">
      <router-link-stub tag="a" to="[object Object]">
        First folder
      </router-link-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item">
      <oc-button-stub type="button" size="medium" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
        Subfolder
      </oc-button-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item">
      <router-link-stub tag="a" to="[object Object]">
        Deep
      </router-link-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item"><span aria-current="page" tabindex="-1">Deeper ellipsize in responsive mode</span>
      <!---->
    </li>
  </ol>
  <div class="oc-breadcrumb-drop"><label tabindex="0" class="oc-breadcrumb-drop-label"><span aria-current="page" class="oc-breadcrumb-drop-label-text">Deeper ellipsize in responsive mode</span>
      <oc-icon-stub name="expand_more" filltype="fill" accessiblelabel="Expand more" type="span" size="medium" variation="passive" class="oc-breadcrumb-drop-label-icon"></oc-icon-stub>
    </label>
    <oc-drop-stub dropid="oc-drop-1" toggle="" position="bottom-start" mode="click" paddingsize="medium" options="[object Object]">
      <ol>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            Deep
          </router-link-stub>
        </li>
        <li>
          <oc-button-stub type="button" size="medium" variation="passive" appearance="raw" justifycontent="left" gapsize="medium">
            Subfolder
          </oc-button-stub>
        </li>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            First folder
          </router-link-stub>
        </li>
      </ol>
    </oc-drop-stub>
  </div>
</nav>
`;
